server/game/cards/attachments/04/beggarking.js

```diff
@@ -9,10 +9,10 @@ class BeggarKing extends DrawCard {
         });
         this.reaction({
             when: {
-                onPlotRevealCompleted: () => (
-                    _.any(this.game.getPlayers(), player => (
-                        player !== this.controller &&
-                        this.controller.activePlot.getIncome(true) < player.activePlot.getIncome(true)
+                onPlotsRevealed: event => (
```

Comment:
The logic is more correct now since it's only looking at the plots revealed instead of the plots for all players (e.g. Rains where only one person is revealing a plot), but I agree it's ugly.

I generally try to avoid code that assumes only 2 players in case we find a good way to support Melee but we can worry about that later. I've updated Beggar King in the interest of clarity.